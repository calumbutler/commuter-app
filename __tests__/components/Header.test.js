import renderer from 'react-test-renderer';
import Header from 'js/components/Header';
import React from 'react';

/**
* Snapshot testing can be very useful if used correctly, this example just shows how to use
* it but this is not the most useful snapshot, sometimes it is better to call it multiple
* times inside a test and take snapshots of the component changing throughout some interactions,
* Ex. snapshot default state, snapshot after click/mouseover, snapshot after mouseleave
*/
describe('Header snapshot Tests', () => {

  test('Sample of snapshot testing, not very useful, but shows how to do it', () => {
    const header = renderer.create(
      <Header title="Snapshot Test #1"/>
    );
    const tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
