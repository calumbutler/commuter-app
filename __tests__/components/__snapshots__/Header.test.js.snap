// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot Tests Sample of snapshot testing, not very useful, but shows how to do it 1`] = `
<div
  className="app-header"
>
  <h1
    className="app-title"
  >
    Snapshot Test #1
  </h1>
  <h2
    className="app-subtitle"
  >
    Test output over time in your component
  </h2>
  <img
    alt="app-logo"
    className="app-logo"
    src={Object {}}
  />
</div>
`;
